2 BE and 3 FE tests added, and Pages are re-organized with PEP8 guidelines

Rohanksingh18 opened the pull request:
Kindly review.

Rohanksingh18 committed:
2 BE and 3 FE tests added, and Pages are re-organized with PEP8 guidelines

Rohanksingh18 committed:
important changes in setup files

Rohanksingh18 committed:
Delete README.md

committed mistekenly

Rohanksingh18 committed:
Delete desktop.ini

created mistakenly

Rohanksingh18 committed:
3 more BE test added with gitlab configuration

Rohanksingh18 committed:
Merge branch 'wip' of https://github.com/Rohanksingh18/ecom_site_automation into wip

Rohanksingh18 committed:
five more BE tests added to the framework

admaskinfu commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,53 @@
+    run_tests
+  image: python:3.11
+  script:
+    - echo "STRING BACKEND TESTS....."
```
You mean 'Starting'? typo

admaskinfu commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,53 @@
+    paths:
+      - ./output
+
+run_ecom_smoke_frontend:
```
come up with a better name.
May be have 2 jobs (run_ecom_smoke_frontend, run_ecom_regression_frontend, run_ecom_smoke_backend, run_ecom_regression_backend)

Rohanksingh18 replied:
Changed.

Rohanksingh18 replied:
suggestions applied

admaskinfu commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,53 @@
+    - name: selenium/standalone-chrome:4.7.1-20221208
+      alias: chrome_remote_browser
+  script:
+    - echo "STRING FRONTEND SMOKE TESTS....."
```
typo

Rohanksingh18 replied:
rectified

admaskinfu commented on README.md:
Why is this deleted?

Rohanksingh18 replied:
Created here by mistake

admaskinfu commented on ecom_test/src/api_helpers/CouponsAPIHelper.py:
```diff
@@ -0,0 +1,29 @@
+
+    # calling the get method of the woo_api_utility instance.
+    def call_list_all_coupons(self, payload=None):
+        logger.debug("Calling list all coupons.")
```
the messaging should be more readable like "Calling 'list all coupons'." Just adding the inside quotes.

Rohanksingh18 replied:
changed

admaskinfu commented on ecom_test/src/api_helpers/CouponsAPIHelper.py:
```diff
@@ -0,0 +1,29 @@
+    def __init__(self):
+        self.woo_api_utility = WooAPIUtility()
+
+    # Method makes a POST request to create a new coupon by calling the post-method of the woo_api_utility instance.
```
Use proper docstring and not comments. You MUST use doc strings. Under the function name use 3 quotes to add doc string. do a quick look up of doc strings.
Example
`def call_create_coupon(self, payload):
       """
       Makes a POST request to create a new coupon by calling the post-method of the woo_api_utility instance.
       """`

Rohanksingh18 replied:
changed

admaskinfu commented on ecom_test/src/api_helpers/ProductsAPIHelpers.py:
```diff
@@ -2,8 +2,8 @@
 
-#Create a class for product API
 
+# Create a class for product API
```
Add it as doc string not comment.

Rohanksingh18 replied:
done

admaskinfu commented on ecom_test/src/pages/HomePage.py:
```diff
@@ -19,5 +17,11 @@ def go_to_the_home_page(self):
         self.sl.wait_and_click(self.Add_To_Cart_BTN)
 
+    def get_all_product_elements(self):
+        return self.sl.wait_and_get_elements(self.Product_Elements)
```
For the locator variables, use upper case letter. It is a convention to use upper case for constants. Those are variables that will not change so they are constants. It looks much better.
If you do not agree with that then make all lower case. Definitely do not do 'Product_Elements'.

Rohanksingh18 replied:
made change for every locators

admaskinfu commented on ecom_test/src/pages/locators/MyAccountSignedInPageLocators.py:
```diff
@@ -1,5 +1,6 @@
 
+
 class MyAccountSignedInPageLocator:
     Left_Nav_LogOut_BTN = (By.XPATH, "//a[normalize-space()='Logout']") #to verfiy logOut button is visbile after registration
```
Use upper case for the constants (these variables are constant)

Rohanksingh18 replied:
changed

admaskinfu commented on ecom_test/tests/backend_tests/coupons/test_create_coupons.py:
```diff
@@ -0,0 +1,98 @@
+
+@pytest.fixture(scope='module')
+def my_setup():
+    """
```
This is a good example of using doc string.

Rohanksingh18 replied:
Thanks

admaskinfu commented on variables.sh:
```diff
@@ -1,11 +1,9 @@
-set WOO_KEY=ck_b7504831538d902f98ebdb3c97e541412b1c3970
-set WOO_SECRET=cs_778cd37bb50c506ba42f964b9094a37cc21af3a3
\ No newline at end of file
+export BASE_URL=http://localhost:8888/EcomSite/
```
You can keep 2 variable files. One for your local (Windows version) and one for Linux version.

Rohanksingh18 replied:
Yes, I have but it's only in the local only as I added it into gitignore. However, I will push that well.

Rohanksingh18 committed:
Update .gitlab-ci.yml

typo mistakes are rectified and suggestions are implemented in the code for 'gitlab-ci.yml" file.

Rohanksingh18 committed:
Update CouponsAPIHelper.py

Changes have been made

Rohanksingh18 committed:
Update HomePage.py

rectified

Rohanksingh18 committed:
Update ProductsAPIHelpers.py

Rectified

Rohanksingh18 committed:
Update MyAccountSignedInPageLocators.py

All locators are updated now..

Rohanksingh18 committed:
changes requests

Rohanksingh18 commented on README.md:
done

Rohanksingh18 commented on desktop.ini:
done

Rohanksingh18 committed:
Resolve merge conflicts

Rohanksingh18 committed:
Resolve merge conflicts

Rohanksingh18 commented:
All done
